product_cost_price_avco_sync/models/purchase_order.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2019 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    def write(self, vals):
+        res = super().write(vals)
+        if 'price_unit' in vals and not self.env.context.get(
```

Comment:
You should take into account other values that will vary the final price unit, like taxes. Purchase discount modification will also modify the move price unit. Although not performant, the only way I see for not needing a lot of glue modules is that you call `_get_stock_move_price_unit` before calling write records per record, and again after, and propagate the price unit if changed.